libc/src/__support/FPUtil/dyadic_float.h

```diff
@@ -183,6 +184,10 @@ template <size_t Bits> struct DyadicFloat {
     return r;
   }
 
+  LIBC_INLINE explicit constexpr operator float16() const {
+    return static_cast<float16>(static_cast<float>(*this));
+  }
```

Comment:
The smoke test for `ldexpf16` showed that `operator T()` defined above did not always give correct results for `float16`. At least this part did not: https://github.com/llvm/llvm-project/blob/main/libc/src/__support/FPUtil/dyadic_float.h#L160-L161.

Maybe we could have an exhaustive test for `ldexpf16`.